File: backend/src/main/java/org/course/ecommerce/dto/OrderDetailsDto.java
package org.course.ecommerce.dto;

import java.math.BigDecimal;

public class OrderDetailsDto {

    private Integer orderId;
    private Integer productId;
    private String productName;
    private BigDecimal unitPrice;
    private Short quantity;
    private Float discount;

    public OrderDetailsDto() {}

    public Integer getOrderId() { return orderId; }
    public void setOrderId(Integer orderId) { this.orderId = orderId; }

    public Integer getProductId() { return productId; }
    public void setProductId(Integer productId) { this.productId = productId; }

    public String getProductName() { return productName; }
    public void setProductName(String productName) { this.productName = productName; }

    public BigDecimal getUnitPrice() { return unitPrice; }
    public void setUnitPrice(BigDecimal unitPrice) { this.unitPrice = unitPrice; }

    public Short getQuantity() { return quantity; }
    public void setQuantity(Short quantity) { this.quantity = quantity; }

    public Float getDiscount() { return discount; }
    public void setDiscount(Float discount) { this.discount = discount; }
}

File: mobile/app/src/main/java/org/course/mobile/data/network/dto/OrderDetailsDto.kt
package org.course.mobile.data.network.dto

import kotlinx.serialization.Serializable

// Mirrors org.course.ecommerce.dto.OrderDetailsDto. orderId+productId together are
// the composite primary key server-side (OrderDetailsId) - both required to
// identify a specific line for the .../{orderId}/{productId} endpoints.
@Serializable
data class OrderDetailsDto(
    val orderId: Int,
    val productId: Int,
    val productName: String? = null,
    val unitPrice: Double? = null,
    val quantity: Short? = null,
    val discount: Float? = null,
)
